PencilCase/PencilCase/scripts/Navigator.ts

```diff
@@ -0,0 +1,104 @@
+    }
+
+    public goHome = () => {
+        Application.instance.activePage(new HomePage());
+        $(':mobile-pagecontainer').pagecontainer("change", "#HomePage");
+    }
+
+    public initialize = () => {
+        $(':mobile-pagecontainer').on("pagecontainerbeforechange", (eventObject: JQueryEventObject, parameters: any) => {
+            if (parameters.toPage !== Consts.Pages.HomePage.Id) { // No need the handling here for home page
+                if ((parameters.options && parameters.options.data)) {
+                    let data = parameters.options.data;
+                    let pp = Application.instance.activePage();
+                    if (Application.instance.activePage().pageId !== data.pageInfo.Id) {
+                        // Since this page before change event will be called 2 times, so add code here to avoid set active page 2 times
+                        let page = this.getPage(data.pageInfo);
+                        if (data.refresh) {
+                            // If have refresh parameter and value is true, refresh the target page
+                            page.initialize();
+                        }
+
+                        Application.instance.activePage(page);
+                    }
+                }
+            }
+            else {
+                Application.instance.activePage(Application.instance.homePage());
+            }
+        });
+
+        $(':mobile-pagecontainer').pagecontainer({
+            beforeshow: (eventObject: JQueryEventObject, ui: any) => {
+                if (!Application.instance.activePage().equals(Application.instance.homePage())) {
+                    // http://demos.jquerymobile.com/1.3.2/faq/injected-content-is-not-enhanced.html
+                    $("body").pagecontainer("getActivePage").trigger("create");
+                }
+            }
+        });
+    }
+
+    private getPage(pageInfo: any) {
+        let page: PageBase;
+        let pageExisted: boolean = false;
+        switch (pageInfo) {
+            case Consts.Pages.ProductManagement:
+                page = this.getExistedInstance(pageInfo);
+                pageExisted = !(page == null);
+                if (pageExisted == false)
+                    page = new ProductManagement();
+                break;
+            case Consts.Pages.Retail:
+                page = this.getExistedInstance(pageInfo);
+                pageExisted = !(page == null);
+                if (pageExisted == false)
+                    page = new Retail();
+                break;
+        }
+
+        if (pageExisted == false && pageInfo.IsPermanent === true)
+            Application.instance.pages.push(page);
+
+        return page;
+    }
+
+    private getExistedInstance(pageInfo: any) {
+        if (pageInfo.IsPermanent == false) return null;
+
+        for (let i = 0; i < Application.instance.pages.length; i++) {
+            if (Application.instance.pages[i].pageId === pageInfo.Id) {
```

Comment:
Plan to set Retail, Wholesale page as permanent pages, put them into Application pages collection, keeping them in memory. 
For home page, store it into application homePage fields.